iree/compiler/Dialect/HAL/Conversion/HALToVM/test/buffer_ops.mlir

```diff
@@ -1,54 +1,147 @@
+  // CHECK-DAG: %[[HI_I64:.+]] = vm.shl.i64 %[[HI_EXT]], %c32
+
+  // CHECK-DAG: %[[LO_I32:.+]] = vm.call @hal.buffer.load(%[[BUFFER]], %c64, %c4) : (!vm.ref<!hal.buffer>, i32, i32) -> i32
+  // CHECK-DAG: %[[LO_I64:.+]] = vm.ext.i32.i64.u %[[LO_I32]]
+
+  // CHECK: %[[RET:.+]] = vm.or.i64 %[[LO_I64]], %[[HI_I64]]
+
+  %0 = hal.buffer.load<%buffer: !hal.buffer>[%c64] : i64
+  // CHECK: return %[[RET]]
+  return %0 : i64
+}
+
+// -----
+
+// CHECK-LABEL: @buffer_load_f32
+// CHECK-SAME: (%[[BUFFER:.+]]: !vm.ref<!hal.buffer>)
+func @buffer_load_f32(%buffer: !hal.buffer) -> f32 {
+  %c64 = arith.constant 64 : index
+  // CHECK: %[[RET_I32:.+]] = vm.call @hal.buffer.load(%[[BUFFER]], %c64, %c4) : (!vm.ref<!hal.buffer>, i32, i32) -> i32
+  %0 = hal.buffer.load<%buffer: !hal.buffer>[%c64] : f32
+  // CHECK: %[[RET:.+]] = vm.bitcast.i32.f32 %[[RET_I32]]
+  // CHECK: return %[[RET]]
+  return %0 : f32
+}
+
+// -----
+
+// CHECK-LABEL: @buffer_store_i8
+// CHECK-SAME: (%[[BUFFER:.+]]: !vm.ref<!hal.buffer>, %[[VALUE:.+]]: i32)
+func @buffer_store_i8(%buffer: !hal.buffer, %value: i8) {
+  %c64 = arith.constant 64 : index
+  // CHECK: vm.call @hal.buffer.store(%[[VALUE]], %[[BUFFER]], %c64, %c1) : (i32, !vm.ref<!hal.buffer>, i32, i32) -> ()
+  hal.buffer.store<%buffer : !hal.buffer>[%c64] value(%value : i8)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @buffer_store_i16
+// CHECK-SAME: (%[[BUFFER:.+]]: !vm.ref<!hal.buffer>, %[[VALUE:.+]]: i32)
+func @buffer_store_i16(%buffer: !hal.buffer, %value: i16) {
+  %c64 = arith.constant 64 : index
+  // CHECK: vm.call @hal.buffer.store(%[[VALUE]], %[[BUFFER]], %c64, %c2) : (i32, !vm.ref<!hal.buffer>, i32, i32) -> ()
+  hal.buffer.store<%buffer : !hal.buffer>[%c64] value(%value : i16)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @buffer_store_i32
+// CHECK-SAME: (%[[BUFFER:.+]]: !vm.ref<!hal.buffer>, %[[VALUE:.+]]: i32)
+func @buffer_store_i32(%buffer: !hal.buffer, %value: i32) {
+  %c64 = arith.constant 64 : index
+  // CHECK: vm.call @hal.buffer.store(%[[VALUE]], %[[BUFFER]], %c64, %c4) : (i32, !vm.ref<!hal.buffer>, i32, i32) -> ()
+  hal.buffer.store<%buffer : !hal.buffer>[%c64] value(%value : i32)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @buffer_store_i64
+// CHECK-SAME: (%[[BUFFER:.+]]: !vm.ref<!hal.buffer>, %[[VALUE:.+]]: i64)
+func @buffer_store_i64(%buffer: !hal.buffer, %value: i64) {
+  %c64 = arith.constant 64 : index
+
+  // CHECK-DAG: %[[VALUE_LO:.+]] = vm.trunc.i64.i32 %[[VALUE]]
+  // CHECK: vm.call @hal.buffer.store(%[[VALUE_LO]], %[[BUFFER]], %c64, %c4) : (i32, !vm.ref<!hal.buffer>, i32, i32) -> ()
+
+  // CHECK-DAG: %[[VALUE_HI_I64:.+]] = vm.shr.i64.u %[[VALUE]], %c32
+  // CHECK-DAG: %[[VALUE_HI:.+]] = vm.trunc.i64.i32 %[[VALUE_HI_I64]]
+  // CHECK-DAG: %[[OFFSET_HI:.+]] = vm.add.i32 %c64, %c4
+  // CHECK: vm.call @hal.buffer.store(%[[VALUE_HI]], %[[BUFFER]], %[[OFFSET_HI]], %c4) : (i32, !vm.ref<!hal.buffer>, i32, i32) -> ()
+
+  hal.buffer.store<%buffer : !hal.buffer>[%c64] value(%value : i64)
```

Comment:
Should there be any endianness considerations in this new code?

Reply:
nah, we're little endian only all over the place - dealing with endianness mismatch between host/device would be extremely difficult and no one should be using big endian anymore (especially not in ML). No one today is writing out models with their constants byte flipped to big endian for example :P